protocol/version.go

```diff
@@ -9,8 +9,10 @@ const (
 	Version36
 	Version37
 	Version38
+	Version39
 	VersionWhatever    VersionNumber = 0 // for when the version doesn't matter
 	VersionUnsupported VersionNumber = -1
+	VersionUnknown     VersionNumber = -2
```

Comment:
`VersionWhatever` is only intended for tests (although it's also used for writing the Version Negotiation Packet). We could consider moving that to the test code, so that it's not available in the production code anymore (if Go allows that).
I don't think that we should get rid of `VersionUnsupported`, since this is used for doing the version negotiation on the client side.

I'm not sure what the right solution here is. Maybe keep `VersionWhatever` and `VersionUnsupported`, and use the bool instead of `VersionUnknown`, as you suggested. What do you think?